owiddata/04.drawMaps.py

```diff
@@ -0,0 +1,143 @@
+        else:
+            fixedlist.append(iso)
+
+
+def setup_geopandas():
+    countries_mapping = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))
+    countries_mapping = lowres_fix(countries_mapping)
+    countries_mapping = countries_mapping[(countries_mapping.name != "Antarctica") &
+                                          (countries_mapping.iso_a3 != "-99")]
+    return countries_mapping
+
+def main(args):
+    # Set up country mapping
+    countries_mapping = setup_geopandas()
+
+    # Load existing stats & grab scale for map from JSON
+    owid_stats = load_JSON(args.update_json)
+
+    # Load the integrated OWID/VIPER data
+    vaxPlatforms = pd.read_csv(args.platform_types)
+
+    # Count the number of vaccines of each type that OWID has data for
+    # Then find max
+    platformCounts = vaxPlatforms[vaxPlatforms.countries.notnull()].\
+        groupby("Platform").size()
+    maxNumVax = max(platformCounts)
+
+    # Set the parameters color-coding the plots. Scale is the max candidates
+    # adminstered across all vaccine types in the OWID data
+    scale = maxNumVax + 1
+    cmap = mpl.cm.Purples
+    norm = mpl.colors.BoundaryNorm(np.arange(0, scale), cmap.N)
+
+    for platform in set(vaxPlatforms["Platform"]):
+        platformName = '_'.join(platform.split(' '))
+        platformName = platformName.replace("-", "_")
+        vaccines = vaxPlatforms[vaxPlatforms["Platform"] == platform].dropna()
+
+        # This stat does not reveal the overall number of approved vaccines,
+        # just the number tracked by OWID
+        owid_stats["owid_" + platformName + "_count"] = \
+            len(vaccines)
+        if len(vaccines) == 0:
+            owid_stats["owid_" + platformName + "_countries"] = len(vaccines)
+            print("No data in OWID dataset for {}".format(platform))
+            continue
+
+        # Split lists in the "countries" column into individual ISO codes
+        # and count how many unique countries appear
+        listOfCodes= vaccines["countries"].apply(literal_eval)
+        countries = pd.Series([iso for country_list in listOfCodes
+                               for iso in country_list])
+        owid_stats["owid_" + platformName + "_countries"] = len(countries.unique())
+
+        # Count how many times a country code appears across all vaccines
+        # of this platform type
+        vaxPresence = pd.DataFrame(countries.value_counts())
+        vaxPresence.rename({0:platform}, axis=1, inplace=True)
+
+        # Use the vaxPresence data to set up the chloropeth
+        mappingData = countries_mapping.merge(vaxPresence,
+                                              how="left",
+                                              right_index=True,
+                                              left_on="iso_a3")
+        mappingData[platform] = mappingData[platform].fillna(0)
+
+        # plot data
+        fig, ax = plt.subplots(1, 1, figsize=(6,4))
```

Comment:
Since already been implemented feel free to ignore but might want to also look into [plotnine](https://plotnine.readthedocs.io). R's ggplot ported into python. Plus it works with plydata.